ci-automation/sdk_container_update.sh

```diff
@@ -0,0 +1,109 @@
+#
+# sdk_container_update() should be called w/ the positional INPUT parameters below.
+
+# SDK container update automation stub.
+#  This script will create a new SDK container image based on an existing container image.
+#  It is meant to be used for updating tools in the SDK that do not directly affect OS image
+#  generation - like e.g. the mantle suite tools (kola, ore, etc.).
+#
+# PREREQUISITES:
+#
+#   1. SDK version is recorded in sdk_container/.repo/manifests/version.txt and a matching source
+#       SDK container is available on build cache at "/containers/[VERSION]/flatcar-sdk-all-[VERSION].tar.gz"
+#         (nightly / dev builds)
+#      OR
+#       SDK image is available on ghcr.io/flatcar-linux/flatcar-sdk-all:[VERSION]
+#         (official SDK releases)
+#      Specifically, the "-all" version of the source SDK image is required.
+#
+# OPTIONAL INPUT
+#
+#   1. Target SDK version number (without alpha-/beta-/etc. prefix).
+#      If not provided it will be computed from the source SDK, and the patch level is bumped
+#      (e.g. 3139.0.0 => 3139.0.1).
+#
+#   2. coreos-overlay repository tag to use (commit-ish).
+#       Optional - use scripts repo sub-modules as-is if not set.
+#       This version will be checked out / pulled from remote in the coreos-overlay git submodule.
+#       The submodule config will be updated to point to this version before the TARGET SDK tag is created and pushed.
+#
+#   3. portage-stable repository tag to use (commit-ish).
+#       Optional - use scripts repo sub-modules as-is if not set.
+#       This version will be checked out / pulled from remote in the portage-stable git submodule.
+#       The submodule config will be updated to point to this version before the TARGET SDK tag is created and pushed.
+#
+# OUTPUT:
+#
+#   1. SDK container image of the new SDK, published to buildcache
+#       at "/containers/[NEW_VERSION]/flatcar-sdk-all-[NEW_VERSION].tar.gz"
+#
+#   2. New SDK version tagged in scripts and tag pushed to origin.
+#
+#   3. "./ci-cleanup.sh" with commands to clean up temporary build resources,
+#        to be run after this step finishes / when this step is aborted.
+
+set -euo pipefail
+
+# Helper function to generate a new version from the current SDK version
+#  in version.txt.
+# Can also be used by CI automation to generate "$1" for sdk_container_update
+#  coreos_git and portage_git are to be provided.
+#
+function sdk_container_update_generate_new_version() {
+    source ci-automation/ci_automation_common.sh
+    source sdk_container/.repo/manifests/version.txt
+
+    local current_vernum="${FLATCAR_SDK_VERSION}"
+
+    local current_major_minor="$(echo "${current_vernum}" | sed 's/^\([0-9]\+\.[0-9]\+\).*/\1/')"
+    local current_patchlevel="$(echo "${current_vernum}" | sed 's/^[0-9]\+\.[0-9]\+\.\([0-9]\+\).*/\1/')"
+    local current_suffix="$(echo "${current_vernum}" | sed 's/^\([0-9]\+\.[0-9]\+\.[0-9]\+\)//')"
+
+    current_patchlevel="$((current_patchlevel + 1))"
```

Comment:
how do you prevent collisions here?

Reply:
I'd argue it's out of scope for this script and should be ensured by whoever called the script. When in doubt there's always the `version` parameter to `sdk_container_update` ([line 75](https://github.com/flatcar-linux/scripts/pull/240/files/fae06c6d7fe173a3eb55e2040fd3eebfc7b77c0b#diff-19c30e9a3870cff9d8ab067cdbac510a9b281976406c1f54c2ae9ba718367fbcR75)).

Reply:
But the change is pushed at the very last line of `update_sdk_container`, which means the step succeeded. Why would we retrigger a step that succeeded?

Reply:
To get another mantle update.

Reply:
OK, I can now follow this and that there is a way for the user to override the version.